File: FrontEnd/GranjaSA/src/app/main/Domain/Cliente.ts
import { Porcino } from "./Porcino";

export class Cliente {
    cedula: number;
    nombres: string;
    apellidos: string;
    direccion: string;
    telefono: string;
    porcinos?: Porcino[]

    constructor(cedula: number, nombres: string, apellidos: string, direccion: string, telefono: string, porcinos?: Porcino[]) {
        this.cedula = cedula;
        this.nombres = nombres;
        this.apellidos = apellidos;
        this.direccion = direccion;
        this.telefono = telefono;
        this.porcinos = porcinos;
    }
}

File: BackEnd/src/main/java/com/example/granjasa/Entity/Cliente.java
package com.example.granjasa.Entity;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;

import com.fasterxml.jackson.annotation.JsonIgnore;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.Data;

@Entity
@Data
@Table(name = "cliente")
public class Cliente implements Serializable {

    @Id
    private Integer cedula;

    @NotNull(message = "Los nombres son obligatorios")
    @Column(nullable = false)
    private String nombres;

    @NotNull(message = "Los apellidos son obligatorios")
    @Column(nullable = false)
    private String apellidos;

    @NotNull(message = "La dirección es obligatoria")
    @Column(nullable = false)
    private String direccion;

    @NotNull(message = "El teléfono es obligatorio")
    @Column(nullable = false)
    private String telefono;

    @OneToMany(mappedBy = "cliente", cascade = {CascadeType.PERSIST, CascadeType.MERGE})
    @JsonIgnore
    private List<Porcino> porcinos = new ArrayList<>();
    
}
